internal/cmd/gocdk/main.go

```diff
@@ -124,6 +138,38 @@ func (pctx *processContext) gcpCredentials(ctx context.Context) (*google.Credent
 	return gcp.DefaultCredentials(ctx)
 }
 
+func (pctx *processContext) Logf(format string, a ...interface{}) {
+	pctx.errlog.Printf(format, a...)
+}
+
+func (pctx *processContext) Printf(format string, a ...interface{}) {
+	pctx.outlog.Printf(format, a...)
+}
+
+func (pctx *processContext) Println(a ...interface{}) {
+	pctx.outlog.Println(a...)
+}
+
+// ApplyToCmd updates fields of cmd for this processContext.
+//
+// dir may be empty; it defaults to pctx.workdir.
+//
+// Note that this function sets cmd.Stdout, and that Cmd.Output requires
+// Stdout to be nil; you may need to set it back to nil explicitly (or
+// don't use this function). Similarly for Cmd.CombinedOutput, for both
+// Stdout and Stderr.
+func (pctx *processContext) ApplyToCmd(cmd *exec.Cmd, dir string) {
```

Comment:
Nice, I've been thinking of creating a similar function for a while.

"Apply" didn't evoke for me what this function actually does. What if we rearranged this to have the same signature as `os/exec.CommandContext`, but filling in all the fields like you have? Then this would be `pctx.NewCommand(ctx, "terraform", "output")` etc.

Reply:
Nice, I like it. Done.